components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -375,10 +430,56 @@ public void run() {
                 EmailEventAdapterUtil.logAndDrop(eventAdapterConfiguration.getName(), message,
                         "Error in message format", e, log, tenantId);
             } catch (Exception e) {
-                EventAdapterUtil.logAndDrop(eventAdapterConfiguration.getName(), message, "Error sending email to '" + to + "'", e, log, tenantId);
+                EventAdapterUtil.logAndDrop(eventAdapterConfiguration.getName(), message,
+                        "Error sending email to '" + to + "'", e, log, tenantId);
+            }
+        }
+    }
+
+    private void sendWithRetry(Message message) throws MessagingException {
+
+
+        try {
+            Transport.send(message);
+        } catch (AuthenticationFailedException exception) {
+            // Retry only for CLIENT_CREDENTIAL authentication type.
+            if (!CLIENT_CREDENTIAL.equalsIgnoreCase(globalProperties.get(ADAPTER_EMAIL_AUTH_TYPE))) {
+                throw new AuthenticationFailedException();
             }
+            handleRetry(message);
+            log.warn("Authentication failed, attempting token refresh and retry...", exception);
```

Comment:
Can the server be exhaused with this exception log, if an attacker/script tries this flow continuously?

Reply:
Add it as a debug log